File: test-case/list/revall/revall.php
<?php
require_once 'Soda/DS/index.php';
require_once 'Soda/Leetcode/index.php';
require_once 'Soda/Unittest/index.php';

use Soda\DS;
use Soda\Leetcode\ListNode;
use Soda\Unittest\TestWork;
use Soda\Unittest\Utils;
use Soda\Unittest\Validators;

class Solution {

    /**
     * @param ListNode[] $lists
     * @return ListNode[]
     */
    function reverseAll($lists) {
        for ($i = 0; $i < count($lists); ++$i) {
            $lists[$i] = $this->reverse($lists[$i]);
        }
        [$i, $j] = [0, count($lists)-1];
        while ($i < $j) {
            [$lists[$i], $lists[$j]] = [$lists[$j], $lists[$i]];
            ++$i;
            --$j;
        }
        return $lists;
    }

    private function reverse($head) {
        $h = null;
        while ($head) {
            $next = $head->next;
            $head->next = $h;
            $h = $head;
            $head = $next;
        }
        return $h;
    }
}

$work = TestWork::create(new Solution(), 'reverseAll', $argv[0]);
// $work->validator = fn($e, $r) => <boolean result>
$work->compareSerial = true;
echo $work->run(Utils::fromStdin());

File: test-case/list/revall/Revall.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.TestWork;


class Solution {
    public List<ListNode> reverseAll(List<ListNode> lists) {
        var res = new ArrayList<ListNode>();
        for (int i = 0; i < lists.size(); ++i) {
            res.add(reverse(lists.get(i)));
        }
        int i = 0, j = res.size() - 1;
        while (i < j) {
            var temp = res.get(i);
            res.set(i, res.get(j));
            res.set(j, temp);
            ++i;
            --j;
        }
        return res;
    }

    private ListNode reverse(ListNode head) {
        ListNode h = null;
        while (head != null) {
            var next = head.next;
            head.next = h;
            h = head;
            head = next;
        }
        return h;
    }
}

public class Revall implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "reverseAll");
        // var work = TestWork.forStruct(Struct.class);
        // work.setValidator((e, r) -> {...});
        work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Revall().get().run();
    }

}
